corehq/apps/hqmedia/view_helpers.py

```diff
@@ -114,18 +115,42 @@ def update_multimedia_paths(app, paths):
 
 def download_audio_translator_files(domain, app, lang):
     # Get bulk app translation single sheet data
+    headers = get_bulk_app_sheet_headers(app, single_sheet=True, lang=lang, eligible_for_transifex_only=True)
+    headers = headers[0]    # There's only one row since these are the headers for the single-sheet format
+    headers = headers[1]    # Drop the first element (sheet name), leaving the second (list of header names)
+    text_index = headers.index('default_' + lang)
+    audio_index = headers.index('audio_' + lang)
     sheets = get_bulk_app_single_sheet_by_name(app, lang, eligible_for_transifex_only=True)
-    translations = sheets[SINGLE_SHEET_NAME]
-
-    # Get missing multimedia data
-    missing_media_rows = download_multimedia_paths_rows(app, only_missing=True)
-    missing_media_paths = [row[0] for (sheet_name, row) in missing_media_rows]
+    audio_rows = [row for row in sheets[SINGLE_SHEET_NAME] if row[audio_index]]
+
+    # Create dict of rows, keyed by label text for de-duplication
+    rows_by_text = defaultdict(list)
+    for row in audio_rows:
+        rows_by_text[row[text_index]].append(row)
+
+    # Create file for translator
+    translator_workbook = openpyxl.Workbook()
+    sheet0 = translator_workbook.worksheets[0]
+    sheet0.title = "filepaths"
+    sheet0.append([lang, "audio"])
+    sheet1 = translator_workbook.create_sheet("verif")
+    sheet1.append(headers)
+
+    def _get_filename_from_duplicate_rows(rows):
+        return rows[0][audio_index]
+
+    # Add a row for each unique text label.
+    for text, rows in rows_by_text.items():
+        if not any([row[audio_index] in app.multimedia_map for row in rows]):
```

Comment:
are we adding missing audio paths?

Reply:
What do you mean by missing? If there's no audio path specified in the bulk app translations download, we ignore that row.

Reply:
Ya, Was confirming my understanding.
The statement above `Add a row for each unique text label` and the logic seems different, hence got confused.